test/k8s-integration/cluster.go

```diff
@@ -269,31 +276,108 @@ func clusterUpGKE(gceZone, gceRegion string, numNodes int, imageType string) err
 		}
 	}
 
-	var cmd *exec.Cmd
-	cmdParams := []string{"container", "clusters", "create", *gkeTestClusterName,
-		locationArg, locationVal, "--num-nodes", strconv.Itoa(numNodes),
-		"--quiet", "--machine-type", "n1-standard-2", "--image-type", imageType}
-	if isVariableSet(gkeClusterVer) {
-		cmdParams = append(cmdParams, "--cluster-version", *gkeClusterVer)
+	if useStagingDriver {
+		accessToken, err := exec.Command("gcloud", "auth", "print-access-token").Output()
+		if err != nil {
+			return fmt.Errorf("failed to get authentication token: %v", err)
+		}
+
+		token := &oauth2.Token{AccessToken: string(accessToken[:len(accessToken)-1])}
+		oauthClient := oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(token))
+		service, err := container.NewService(context.Background(), option.WithHTTPClient(oauthClient))
+		if err != nil {
+			return fmt.Errorf("failed to create a new service: %v", err)
+		}
+
+		service.BasePath = "https://test-container.sandbox.googleapis.com/" // e.g test, staging env URL
```

Comment:
put a todo here to change to read CLOUDSDK_API_ENDPOINT_OVERRIDES_CONTAINER env var.